datafusion/physical-plan/src/sorts/cursor.rs

```diff
@@ -459,4 +560,296 @@ mod tests {
         b.advance();
         assert_eq!(a.cmp(&b), Ordering::Less);
     }
+
+    #[test]
+    fn test_slice_primitive() {
+        let options = SortOptions {
+            descending: false,
+            nulls_first: true,
+        };
+
+        let buffer = ScalarBuffer::from(vec![0, 1, 2, 3]);
+        let mut cursor = new_primitive_cursor(options, buffer, 0);
+
+        // from start
+        let sliced = Cursor::new(cursor.cursor_values().slice(0, 1));
+        let expected = new_primitive_cursor(options, ScalarBuffer::from(vec![0]), 0);
+        assert_eq!(
+            sliced.cmp(&expected),
+            Ordering::Equal,
+            "should slice from start"
+        );
+
+        // with offset
+        let sliced = Cursor::new(cursor.cursor_values().slice(1, 2));
+        let expected = new_primitive_cursor(options, ScalarBuffer::from(vec![1]), 0);
+        assert_eq!(
+            sliced.cmp(&expected),
+            Ordering::Equal,
+            "should slice with offset"
+        );
+
+        // cursor current position != start
+        cursor.advance();
+        let sliced = Cursor::new(cursor.cursor_values().slice(0, 1));
+        let expected = new_primitive_cursor(options, ScalarBuffer::from(vec![0]), 0);
+        assert_eq!(
+            sliced.cmp(&expected),
+            Ordering::Equal,
+            "should ignore current cursor position when sliced"
+        );
+
+        // slicing on a slice (a.k.a. combining offsets)
+        let sliced = Cursor::new(cursor.cursor_values().slice(1, 3));
+        let sliced = Cursor::new(sliced.cursor_values().slice(2, 1));
+        let expected = new_primitive_cursor(options, ScalarBuffer::from(vec![3]), 0);
+        assert_eq!(
+            sliced.cmp(&expected),
+            Ordering::Equal,
+            "should respect previous slice/windowed boundaries, when re-slicing"
+        );
+    }
+
+    #[test]
+    fn test_slice_primitive_nulls_first() {
```

Comment:
What is the reason for also testing with `nulls_first` / `nulls_last`? I didn't see that the semantics of `slice` differ based on the that setting and thus I don't think these tests add much/any new coverage

I suggest either making a loop in `test_slice_primitive` that uses the different `SortOptions` ensuring they produce the same result or just removing these two tests